dlt/destinations/impl/filesystem/filesystem.py

```diff
@@ -468,6 +468,26 @@ def update_stored_schema(
         # externally changed
         return applied_update
 
+    def _get_actual_columns(self, table_name: str) -> List[str]:
+        """Get actual column names from files in storage for regular (non-delta/iceberg) tables
+        or column names from schema"""
+
+        if self.is_open_table("iceberg", table_name):
```

Comment:
this should be merged into `get_storage_tables` which is already present in the filesystem. it will be really helpful.

note: both are operating on Arrow schemas so look how `get_storage_table` works in lance... you do not need to implement TypeMapper just look at lancedb implementation